### PR TITLE
worflows: activity retry policy

### PR DESCRIPTION
**Pull request by famarting**
# Description

first part of https://github.com/dapr/go-sdk/issues/541

also updated one of the existing examples to show how activity retry policies can be used

## Issue reference

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

**Review by cicoyle — APPROVED** [review]
lgtm

File: workflow/activity_context.go [hunk 1]
```diff
@@ -61,6 +71,26 @@ func ActivityRawInput(input string) callActivityOption {
 	}
 }
 
+func RetryPolicy(policy ActivityRetryPolicy) callActivityOption {
```
Comment (mikeee) [inline workflow/activity_context.go:74]: Would this be better suited as `ActivityRetryPolicy`?

File: workflow/activity_context.go [hunk 2]
```diff
@@ -61,6 +71,26 @@ func ActivityRawInput(input string) callActivityOption {
...
+	return func(opts *callActivityOptions) error {
+		opts.retryPolicy = &policy
+		return nil
+	}
+}
+
+func (opts *callActivityOptions) getRetryPolicy() *task.ActivityRetryPolicy {
```
Comment (mikeee) [inline workflow/activity_context.go:81]: As above

**Review by mikeee — COMMENTED** [review]
Looks like a solid implementation of retry policies for activities, just one potential signature refactor to accommodate the future addition of workflow retry policies. Also an ask - would it be possible to implement a simple activity that does continually fail - to test the retry logic?

**Comment by famarting** [comment]
good to go

**Comment by famarting** [comment]
can we merge?

**Comment by mikeee** [comment]
Can you please add a test for the activity to retry - I think it's crucial to validate this so we have some visibility if it breaks

**Review by mikeee — COMMENTED** [review]
Can you please add a test for the activity to retry - I think it's crucial to validate this so we have some visibility if it breaks

File: examples/workflow/main.go
Comment (mikeee) [inline examples/workflow/main.go:1]: The new activity hasn't been registered and isn't being invoked

**Review by mikeee — CHANGES_REQUESTED** [review]
Thanks for adding tests, just need to register the new activity

File: examples/workflow/main.go
Comment (famarting) [reply examples/workflow/main.go:1]: oops 😄

**Review by mikeee — APPROVED** [review]
lgtm

**Pull request merged**